include/secp256k1_silentpayments.h

```diff
@@ -104,6 +104,28 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_p
     const unsigned char *outpoint_smallest36
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(8);
 
+/** Create Silent Payment tweaked public key from public tweak data.
+ *
+ * Given public tweak data (public keys sum and input hash), calculate the
+ * corresponding tweaked public key:
+ *
+ * A_tweaked = input_hash * A_sum
+ *
+ * The resulting data is useful for light clients and silent payment indexes.
+ *
+ *  Returns: 1 if tweaked public key creation was successful. 0 if an error occured.
+ *  Args:              ctx: pointer to a context object
+ *  Out:         A_tweaked: pointer to the resulting tweaked public key
+ *  In:              A_sum: pointer to the public keys sum
+ *              input_hash: pointer to the 32-byte input hash
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_tweaked_pubkey(
```

Comment:
in " silentpayments: add tweaked pubkey creation routine (for light clients / sp index)" (https://github.com/bitcoin-core/secp256k1/pull/1471/commits/842e5bf427d63b86d55aef3c234c05b42c2a6ad7):

If we end up using the `_create_shared_secret_from_public_data` routine, we could also have this be `_create_tweaked_pubkey_from_public_data` where it takes all the same inputs as `_create_shared_secret_from_public_data` and returns `unsigned char *A_tweaked33`. This function would be used by anyone intending to store this data in an index or serve directly to a light client.